lib/src/simple_op_store_model.thrift

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License"),
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+struct RefConflict {
+  1: required list<binary> removes,
+  2: required list<binary> adds,
```

Comment:
Any thoughts on the "required considered harmful" meme?

Reply:
I hope we'll be able to switch these to optional later, but let me check that thrift lets you do that.

Reply:
It seems to work the same way as protos, so it should be fine to make it optional later (https://stackoverflow.com/questions/37558302/how-do-i-make-a-required-thrift-field-optional). That's under the assumption that we don't have old clients reading the repo. It seems likely that old clients would misbehave even if the fields were marked optional and we stopped writing them (because we probably moved the data elsewhere, and they wouldn't know how to read it). We should add some other mechanism for telling old clients that they can't read too-new repos. Mercurial has a `.hg/requires` file for that. If a client sees an entry in that file that they don't understand, they will abort.

I'll mark this resolved, but let me know if you're still concerned about required vs. optional.